### ISSUE TITLE
Flaky test TestRowNumberOperator, TestRowOperators

### ISSUE THREAD

**Issue by findepi** [trinodb/trino#3058]

**Comment by findepi** [comment]
https://github.com/prestosql/presto/pull/3056/checks?check_run_id=497446197

```
2020-03-10T10:05:46.3521898Z 2020-03-10T04:05:46.351-0600	WARN	http-client-memoryManager-10259	io.prestosql.memory.RemoteNodeMemory	Error fetching memory info from http://127.0.0.1:35533/v1/memory: Server refused connection: http://127.0.0.1:35533/v1/memory
2020-03-10T10:05:46.3587498Z 2020-03-10T04:05:46.358-0600	INFO	pool-36-thread-2	io.airlift.bootstrap.LifeCycleManager	Life cycle stopped.
2020-03-10T10:05:49.4400550Z 2020-03-10T04:05:49.439-0600	WARN	pool-36-thread-2	io.prestosql.tests.LogTestDurationListener	Tests from io.prestosql.operator.aggregation.TestRealRegrInterceptAggregation took 10.34m
2020-03-10T10:05:49.4441119Z 2020-03-10T04:05:49.443-0600	WARN	pool-36-thread-2	io.prestosql.tests.LogTestDurationListener	Tests from io.prestosql.operator.aggregation.TestRealRegrSlopeAggregation took 10.34m
2020-03-10T10:05:49.4451392Z 2020-03-10T04:05:49.444-0600	WARN	pool-36-thread-2	io.prestosql.tests.LogTestDurationListener	Tests from io.prestosql.type.TestRealType took 9.41m
2020-03-10T10:06:58.3791697Z [WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/runner/work/presto/presto/presto-main/target/surefire-reports/2020-03-10T09-55-06_691-jvmRun1.dumpstream
2020-03-10T10:07:25.0309145Z [INFO] 
2020-03-10T10:07:25.0309611Z [INFO] Results:
2020-03-10T10:07:25.0324092Z [INFO] 
2020-03-10T10:07:25.0348048Z [INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
2020-03-10T10:07:25.0348554Z [INFO] 
2020-03-10T10:07:25.3761275Z [INFO] ------------------------------------------------------------------------
2020-03-10T10:07:25.3761715Z [INFO] BUILD FAILURE
2020-03-10T10:07:25.3763220Z [INFO] ------------------------------------------------------------------------
2020-03-10T10:07:25.3764644Z [INFO] Total time:  12:26 min
2020-03-10T10:07:25.3766634Z [INFO] Finished at: 2020-03-10T10:07:25Z
2020-03-10T10:07:25.3771208Z [INFO] ------------------------------------------------------------------------
2020-03-10T10:07:25.3779779Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project presto-main: There are test failures.
2020-03-10T10:07:25.3780430Z [ERROR] 
2020-03-10T10:07:25.3781334Z [ERROR] Please refer to /home/runner/work/presto/presto/presto-main/target/surefire-reports for the individual test results.
2020-03-10T10:07:25.3782829Z [ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
2020-03-10T10:07:25.3783603Z [ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
2020-03-10T10:07:25.3792956Z [ERROR] Command was /bin/sh -c cd /home/runner/work/presto/presto/presto-main && /opt/hostedtoolcache/jdk/8.0.242/x64/jre/bin/java -Dfile.encoding=UTF-8 -Xmx2g -Xms2g -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError -XX:-OmitStackTraceInFastThrow -jar /home/runner/work/presto/presto/presto-main/target/surefire/surefirebooter4655909569483025372.jar /home/runner/work/presto/presto/presto-main/target/surefire 2020-03-10T09-55-06_691-jvmRun1 surefire8686609926670201705tmp surefire_01233433522299120388tmp
2020-03-10T10:07:25.3796533Z [ERROR] Error occurred in starting fork, check output in log
2020-03-10T10:07:25.3797183Z [ERROR] Process Exit Code: 3
2020-03-10T10:07:25.3797706Z [ERROR] Crashed tests:
2020-03-10T10:07:25.3798238Z [ERROR] io.prestosql.operator.TestRowNumberOperator
2020-03-10T10:07:25.3798767Z [ERROR] io.prestosql.type.TestRowOperators
2020-03-10T10:07:25.3799359Z [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
2020-03-10T10:07:25.3804957Z [ERROR] Command was /bin/sh -c cd /home/runner/work/presto/presto/presto-main && /opt/hostedtoolcache/jdk/8.0.242/x64/jre/bin/java -Dfile.encoding=UTF-8 -Xmx2g -Xms2g -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError -XX:-OmitStackTraceInFastThrow -jar /home/runner/work/presto/presto/presto-main/target/surefire/surefirebooter4655909569483025372.jar /home/runner/work/presto/presto/presto-main/target/surefire 2020-03-10T09-55-06_691-jvmRun1 surefire8686609926670201705tmp surefire_01233433522299120388tmp
2020-03-10T10:07:25.3805769Z [ERROR] Error occurred in starting fork, check output in log
2020-03-10T10:07:25.3806316Z [ERROR] Process Exit Code: 3
2020-03-10T10:07:25.3806820Z [ERROR] Crashed tests:
2020-03-10T10:07:25.3807404Z [ERROR] io.prestosql.operator.TestRowNumberOperator
2020-03-10T10:07:25.3807938Z [ERROR] io.prestosql.type.TestRowOperators
2020-03-10T10:07:25.3808869Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
2020-03-10T10:07:25.3809464Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
2020-03-10T10:07:25.3810078Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
2020-03-10T10:07:25.3811184Z [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
2020-03-10T10:07:25.3812984Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
2020-03-10T10:07:25.3814643Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
2020-03-10T10:07:25.3816078Z [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
2020-03-10T10:07:25.3817676Z [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
2020-03-10T10:07:25.3818965Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
2020-03-10T10:07:25.3820240Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
2020-03-10T10:07:25.3821529Z [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
2020-03-10T10:07:25.3823513Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
2020-03-10T10:07:25.3825995Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
2020-03-10T10:07:25.3828806Z [ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
2020-03-10T10:07:25.3830798Z [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
2020-03-10T10:07:25.3832356Z [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
2020-03-10T10:07:25.3833940Z [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
2020-03-10T10:07:25.3835482Z [ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
2020-03-10T10:07:25.3836917Z [ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
2020-03-10T10:07:25.3838573Z [ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
2020-03-10T10:07:25.3839600Z [ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
2020-03-10T10:07:25.3845927Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-10T10:07:25.3847046Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-10T10:07:25.3847735Z [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-10T10:07:25.3848701Z [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-03-10T10:07:25.3849937Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
2020-03-10T10:07:25.3850318Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
2020-03-10T10:07:25.3850659Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
2020-03-10T10:07:25.3851066Z [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
2020-03-10T10:07:25.3851491Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-10T10:07:25.3851830Z [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-10T10:07:25.3852198Z [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-10T10:07:25.3852578Z [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-03-10T10:07:25.3853651Z [ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
2020-03-10T10:07:25.3855854Z [ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
2020-03-10T10:07:25.3858032Z [ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
2020-03-10T10:07:25.3858973Z [ERROR] -> [Help 1]
```

-- this is the tail, but the errors start earlier. I didn't investigate where exactly.

**Comment by findepi** [comment]
That's intersting point in the log (leading timestamps truncated)
```
04:05:30.617-0600	INFO	pool-36-thread-1	io.airlift.bootstrap.LifeCycleManager	Life cycle starting...
04:05:30.617-0600	INFO	pool-36-thread-1	io.airlift.bootstrap.LifeCycleManager	Life cycle startup complete. System ready.
04:05:30.618-0600	INFO	pool-36-thread-1	io.prestosql.server.PluginManager	Registering connector tpch
04:05:30.960-0600	INFO	pool-36-thread-1	io.airlift.bootstrap.LifeCycleManager	Life cycle stopping...
04:05:31.024-0600	INFO	pool-36-thread-1	org.eclipse.jetty.server.AbstractConnector	Stopped http@6459840e{HTTP/1.1, (http/1.1, h2c)}{127.0.0.1:42957}
04:05:31.031-0600	WARN	ContinuousTaskStatusFetcher-20200310_100530_00001_urts9.0.0-10208	io.prestosql.server.remotetask.RequestErrorTracker	Error getting task status 20200310_100530_00001_urts9.0.0: java.io.EOFException: HttpConnectionOverHTTP@4b5974d7::SocketChannelEndPoint@7ffd7636{/127.0.0.1:42957<->/127.0.0.1:45392,ISHUT,fill=-,flush=-,to=44/30000}{io=1/0,kio=1,kro=1}->HttpConnectionOverHTTP@4b5974d7(l:/127.0.0.1:45392 <-> r:/127.0.0.1:42957,closed=false)=>HttpChannelOverHTTP@7bc2fd11(exchange=HttpExchange@3ec07c0d req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@1b6c6c2a(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@fb4e5b7{s=START}],recv=HttpReceiverOverHTTP@63f2ab25(rsp=IDLE,failure=null)[HttpParser{s=CLOSED,0 of -1}]]: http://127.0.0.1:42957/v1/task/20200310_100530_00001_urts9.0.0
```

-- we start shutting down (`Life cycle stopping`) because of startup failure (`Life cycle starting`), but we do not say why.
Then, the log is full of numerous exceptions which look like side-effects of failed startup.
But I don't see the actual failure.

@electrum am i reading this correctly?
